openllm-python/src/openllm/models/llama/modelling_ggml_llama.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import annotations
+import logging
+import typing as t
+import openllm
+from .configuration_llama import DEFAULT_PROMPT_TEMPLATE
+from ..._prompt import process_prompt
+
+if t.TYPE_CHECKING: import ctransformers
+logger = logging.getLogger(__name__)
+
+class GGMLLlama(openllm.LLM["ctransformers.AutoModelForCausalLM", "ctransformers.AutoTokenizer"]):
```

Comment:
This is incorrect actually, since the auto class is a metaclass rather than the actual C class itself. We will probably need to stub this with protocol or sth.